launcher/minecraft/mod/Resource.cpp

```diff
@@ -78,20 +96,32 @@ std::pair<int, bool> Resource::compare(const Resource& other, SortType type) con
             removeThePrefix(this_name);
             removeThePrefix(other_name);
 
-            auto compare_result = QString::compare(this_name, other_name, Qt::CaseInsensitive);
-            if (compare_result != 0)
-                return { compare_result, type == SortType::NAME };
-            break;
+            return QString::compare(this_name, other_name, Qt::CaseInsensitive);
         }
         case SortType::DATE:
             if (dateTimeChanged() > other.dateTimeChanged())
-                return { 1, type == SortType::DATE };
+                return 1;
             if (dateTimeChanged() < other.dateTimeChanged())
-                return { -1, type == SortType::DATE };
+                return -1;
+            break;
+        case SortType::SIZE: {
+            auto order_op = order == Qt::SortOrder::AscendingOrder ? 1 : -1;
+            if (this->type() != other.type()) {
+                if (this->type() == ResourceType::FOLDER)
+                    return -1 * order_op;
+                if (other.type() == ResourceType::FOLDER)
+                    return 1 * order_op;
```

Comment:
I don't know if this contradicts what I said but it should probably work like this 
```suggestion
            if (this->type() != other.type()) {
                if (this->type() == ResourceType::FOLDER)
                    return -1;
                if (other.type() == ResourceType::FOLDER)
                    return 1;
```

Reply:
if I do that then I will have the folders move at bottom on decending